gist/client.py

```diff
@@ -227,6 +228,30 @@ def elide(txt, width=terminal_width()):
     return txt
 
 
+def get_value_from_command(string):
+    """Return the value of a config option, potentially by running a command
+
+    When a config option begins with a ``!`` interpret the remaining text as a
+    shell command which when run prints the config option value to stdout.
+    Otherwise return the original string.
+
+    Argumnet:
+        string: value of an option returned from the config file.
+
+    """
+    command = string.lstrip()
+    if command[0] == '!':
+        process = subprocess.Popen(command[1:],
+                                   shell=True,
```

Comment:
If you are going to use `shell=True`, the command should be wrapped in `shlex.quote()` (see https://docs.python.org/3.5/library/subprocess.html#security-considerations), or do not use `shell=True` and use `shlex.split()` on the command.

Reply:
Looks like `shlex.split()` assumes POSIX which could lead to weirdness on Windows, I'll go with `shlex.quote()`.